File: app/src/main/java/com/th/forge/taxiorders/utils/CurrencyParser.java
package com.th.forge.taxiorders.utils;


import java.math.BigDecimal;
import java.text.NumberFormat;
import java.util.Currency;

public class CurrencyParser {
    public static String getFormattedPrice(Integer amount, String currency, String symbol){
        Currency cur = Currency.getInstance(currency);
        NumberFormat formatter = NumberFormat.getInstance();
        BigDecimal bd = new BigDecimal(amount / 100d);
        formatter.setMinimumFractionDigits(2);
        formatter.setMaximumFractionDigits(2);
        formatter.setCurrency(cur);
        return String.format("%s %s",formatter.format(bd),symbol);
    }
}

File: app/src/main/java/com/th/forge/taxiorders/utils/CurrencyParser.kt
package com.th.forge.taxiorders.utils


import java.math.BigDecimal
import java.text.NumberFormat
import java.util.Currency

object CurrencyParser {
    fun getFormattedPrice(amount: Int?, currency: String, symbol: String): String {
        val cur = Currency.getInstance(currency)
        val formatter = NumberFormat.getInstance()
        val bd = BigDecimal(amount!! / 100.0)
        formatter.minimumFractionDigits = 2
        formatter.maximumFractionDigits = 2
        formatter.currency = cur
        return String.format("%s %s", formatter.format(bd), symbol)
    }
}
